common/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -415,7 +413,10 @@ public static DataType tryFindNotNullType(List<DataType> dataTypes) {
             .findFirst().orElse(UNDEFINED);
     }
 
-    public static DataType fromId(Integer id) {
-        return TYPE_REGISTRY.get(id).get();
+    public static DataType<?> fromId(Integer id) {
```

Comment:
This is only used in `UserDefinedFunctionMetaData` to de-serialize the cluster state/stored data. So I think having this kind of search here running over a few couple times is fine.